### ISSUE TITLE
'address list' doesn't list utxos from 'split' 

### ISSUE THREAD

**Issue by keblek** [LLFourn/gun#49]
I first sent myself 100k sats, then I split into two, then I noticed the bug and tried to do it again so I did a "send all" to one address, then split into three again.

```
un -d $(pwd)/test split 30000sat
> This is the transaction that will be broadcast.
> ╔═════╦════════════════════════════════════════════════════════════════════╦══════════════╗
> ║ in  ║ fbca9506d0de7ecad0e0979fb1127349ae19ecb1d2517479b8d794ea3626bf27:0 ║ 0.00 099 678 ║
> ╠═════╬════════════════════════════════════════════════════════════════════╬══════════════╣
> ║     ║ total                                                              ║ 0.00 099 678 ║
> ╠═════╬════════════════════════════════════════════════════════════════════╬══════════════╣
> ║ out ║ tb1qsrjk4wa9z53wehe7j2zc3ysy09lrapc87ee3t5                         ║ 0.00 009 475 ║
> ╠═════╬════════════════════════════════════════════════════════════════════╬══════════════╣
> ║     ║ tb1qqm4eprhj5nhzzhvr2qn2z2f4899nrua9ckmala                         ║ 0.00 030 000 ║
> ╠═════╬════════════════════════════════════════════════════════════════════╬══════════════╣
> ║     ║ tb1q8tul7q9z97eazxxyfpd94m8xn28qskcls4yey2                         ║ 0.00 030 000 ║
> ╠═════╬════════════════════════════════════════════════════════════════════╬══════════════╣
> ║     ║ tb1qgca8xw8c4t0834e655t6vfk3auesjjgdpl9pz6                         ║ 0.00 030 000 ║
> ╠═════╬════════════════════════════════════════════════════════════════════╬══════════════╣
> ║     ║ total                                                              ║ 0.00 099 475 ║
> ╠═════╬════════════════════════════════════════════════════════════════════╬══════════════╣
> ║ fee ║ 1.004 sats/vb                                                      ║ 0.00 000 203 ║
> ╚═════╩════════════════════════════════════════════════════════════════════╩══════════════╝
> Ok [y/n]? y
╔══════╦══════════════════════════════════════════════════════════════════╗
║ txid ║ a2d9d7848c2b711051cd4317088501ed9faa3885232ebbb4addcb73b7ac1ac80 ║
╚══════╩══════════════════════════════════════════════════════════════════╝
```
```
gun -d $(pwd)/test -s address list
syncing wallet with Esplora(EsploraBlockchainConfig { base_url: "https://blockstream.info/testnet/api", concurrency: Some(4), stop_gap: 10 })
╔════════════════════════════════════════════╦═══════╦═══════╗
║ address                                    ║ value ║ utxos ║
╠════════════════════════════════════════════╬═══════╬═══════╣
║ tb1qv04v98ehqtnfgjyey5gaffhlg2jeruvl43m8gn ║ 0     ║ 0     ║
╠════════════════════════════════════════════╬═══════╬═══════╣
║ tb1qla6ll870hzr4p83wv9dxd2yn0psuywme853x5w ║ 0     ║ 0     ║
╚════════════════════════════════════════════╩═══════╩═══════╝

gun -d $(pwd)/test -s tx list
syncing wallet with Esplora(EsploraBlockchainConfig { base_url: "https://blockstream.info/testnet/api", concurrency: Some(4), stop_gap: 10 })
╔══════════════════════════════════════════════════════════════════╦═════════╦═════════════════════╦══════════════╦══════════════╗
║ txid                                                             ║ height  ║ seen                ║ sent         ║ received     ║
╠══════════════════════════════════════════════════════════════════╬═════════╬═════════════════════╬══════════════╬══════════════╣
║ a0296d39399043643f7b3ad351bcf5023f7071015461ce45549e0bb408c2cc7c ║ 2099501 ║ 2021-10-17T18:52:14 ║ 0.00 100 000 ║ 0.00 099 857 ║
╠══════════════════════════════════════════════════════════════════╬═════════╬═════════════════════╬══════════════╬══════════════╣
║ aa0bd640878d2713328670e3aa58b01add70d5e8cc65f54b08a2bda617d32203 ║ 2099498 ║ 2021-10-17T18:29:37 ║ 0            ║ 0.00 100 000 ║
╠══════════════════════════════════════════════════════════════════╬═════════╬═════════════════════╬══════════════╬══════════════╣
║ fbca9506d0de7ecad0e0979fb1127349ae19ecb1d2517479b8d794ea3626bf27 ║ -       ║ -                   ║ 0.00 099 857 ║ 0.00 099 678 ║
╠══════════════════════════════════════════════════════════════════╬═════════╬═════════════════════╬══════════════╬══════════════╣
║ a2d9d7848c2b711051cd4317088501ed9faa3885232ebbb4addcb73b7ac1ac80 ║ -       ║ -                   ║ 0.00 099 678 ║ 0.00 099 475 ║
╚══════════════════════════════════════════════════════════════════╩═════════╩═════════════════════╩══════════════╩══════════════╝
```

**Comment by LLFourn** [comment]
Right so `address list` only shows external addresses so this is working at intended. I think this is how most wallets work (but iirc you are not the first to be confused by this). Split pays to internal addresses. Can you verify they show up in `gun -s utxo list`?

docs and `--help` can at least be improved before closing this issue.

**Comment by LLFourn** [comment]
I think the terminology was introduced by [BIP32]. Basically, external addresses are meant to be targets for receiving coins from other wallets. Internal address are meant to be for this wallet sending coins to itself (which is what split is doing). 

I think it's reasonable to have `--internal` flag which includes internal addresses in the address list. If that flag is set there should also be a column saying whether it's internal or external I guess. I'd also like the derivation index as a side note.

[BIP32]: https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki

**Comment by LLFourn** [comment]
This was fixed by #78 with the ability to list internal addresses
